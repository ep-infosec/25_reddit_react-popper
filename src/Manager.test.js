// @flow
import React from "react";
import { mount } from "enzyme";

// Public API
import { Manager } from ".";

// Private API
import { ManagerContext } from "./Manager";

describe("Manager component", () => {
  it("renders the expected markup", () => {
    const wrapper = mount(
      <Manager>
        <div id="reference" />
        <div id="popper" />
      </Manager>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("provides the related context", () => {
    const Reference = () => null;
    const referenceNode = document.createElement("div");

    const wrapper = mount(
      <Manager>
        <ManagerContext.Consumer>
          {({ getReferenceRef, referenceNode }) => (
            <Reference
              getReferenceRef={getReferenceRef}
              referenceNode={referenceNode}
            />
          )}
        </ManagerContext.Consumer>
      </Manager>
    );

    wrapper.find(Reference).prop("getReferenceRef")(referenceNode);
    wrapper.update();
    expect(wrapper.find(Reference).prop("referenceNode")).toBe(referenceNode);
  });
});

describe("ReferenceNodeContext", () => {
  it("provides proper default values", () => {
    const Reference = () => null;
    const wrapper = mount(
      <div>
        <ManagerContext.Consumer>
          {({ getReferenceRef, referenceNode }) => (
            <Reference
              getReferenceRef={getReferenceRef}
              referenceNode={referenceNode}
            />
          )}
        </ManagerContext.Consumer>
      </div>
    );

    expect(wrapper.find(Reference).prop("getReferenceRef")).toBeUndefined();
  });
});
