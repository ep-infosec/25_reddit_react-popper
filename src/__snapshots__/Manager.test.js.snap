// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manager component renders the expected markup 1`] = `
<Manager>
  <Provider
    value={
      Object {
        "getReferenceRef": [Function],
        "referenceNode": undefined,
      }
    }
  >
    <div
      id="reference"
    />
    <div
      id="popper"
    />
  </Provider>
</Manager>
`;
